clang/include/clang/CConv/ConstraintVariables.h

```diff
@@ -362,6 +362,13 @@ typedef PointerVariableConstraint PVConstraint;
 // Name for function return, for debugging
 #define RETVAR "$ret"
 
+typedef struct {
+  PersistentSourceLoc PL;
+  std::vector<CVarSet> PS;
+  ProgramInfo &I;
```

Comment:
What is `ProgramInfo` here? We only ever have one `ProgramInfo` instance so I don't see why you should (ever) have to save it.

Reply:
Do individual constraint variables have a reference to it? Other methods on Constraint Variables take it as a parameter?

Reply:
I don't know what you are talking about. Don't save it. When you do `constrainConsVarGeq` just use the `ProgramInfo` you have in hand at that time (see below comments).